apps/desktop-wallet/src/modals/AddressSweepModal.tsx

```diff
@@ -45,178 +43,178 @@ import {
-  const [fee, setFee] = useState(BigInt(0))
-  const [builtUnsignedTxs, setBuiltUnsignedTxs] = useState<node.SweepAddressTransaction[]>()
-  const [isLoading, setIsLoading] = useState(false)
-
-  const isConsolidationRedundant = builtUnsignedTxs && builtUnsignedTxs.length === 0
-  const isConsolidationButtonDisabled = !builtUnsignedTxs || builtUnsignedTxs.length === 0
-
-  useEffect(() => {
-    const buildTransactions = async () => {
-      if (!sweepAddresses.from || !sweepAddresses.to) return
-      setIsLoading(true)
-      try {
-        const { unsignedTxs, fees } = await buildSweepTransactions(sweepAddresses.from, sweepAddresses.to.hash)
+const AddressSweepModal = memo(
+  ({ id, addressHash, onSuccessfulSweep, isUtxoConsolidation }: ModalBaseProp & AddressSweepModalProps) => {
+    const { t } = useTranslation()
+    const dispatch = useAppDispatch()
+    const addresses = useAppSelector(selectAllAddresses)
+    const allAddressHashes = useAppSelector(selectAllAddressHashes)
+    const { sendAnalytics } = useAnalytics()
+    const fromAddress = useAppSelector((s) => selectAddressByHash(s, addressHash))
+
+    const toAddressOptions = addressHash ? addresses.filter(({ hash }) => hash !== fromAddress?.hash) : addresses
+    const fromAddressOptions = useAddressesWithBalance()
+
+    const [sweepAddresses, setSweepAddresses] = useState<{ from?: Address; to?: Address }>({
+      from: fromAddress,
+      to: toAddressOptions.length > 0 ? toAddressOptions[0] : fromAddress
+    })
+    const [fee, setFee] = useState(BigInt(0))
+    const [builtUnsignedTxs, setBuiltUnsignedTxs] = useState<node.SweepAddressTransaction[]>()
+    const [isLoading, setIsLoading] = useState(false)
+
+    const isConsolidationRedundant = builtUnsignedTxs && builtUnsignedTxs.length === 0
+    const isConsolidationButtonDisabled = !builtUnsignedTxs || builtUnsignedTxs.length === 0
+
+    useEffect(() => {
+      const buildTransactions = async () => {
+        if (!sweepAddresses.from || !sweepAddresses.to) return
+        setIsLoading(true)
+        try {
+          const { unsignedTxs, fees } = await buildSweepTransactions(sweepAddresses.from, sweepAddresses.to.hash)
+
+          setBuiltUnsignedTxs(unsignedTxs)
+          setFee(fees)
+        } catch (e) {
+          const message = 'Error while building transaction'
+
+          dispatch(transactionBuildFailed(getHumanReadableError(e, t(message))))
+          sendAnalytics({ type: 'error', message })
+        }
+        setIsLoading(false)
+      }
 
-        setBuiltUnsignedTxs(unsignedTxs)
-        setFee(fees)
-      } catch (e) {
-        const message = 'Error while building transaction'
+      buildTransactions()
+    }, [dispatch, sendAnalytics, sweepAddresses.from, sweepAddresses.to, t])
 
-        dispatch(transactionBuildFailed(getHumanReadableError(e, t(message))))
-        sendAnalytics({ type: 'error', message })
-      }
-      setIsLoading(false)
-    }
+    const onAddressChange = useCallback(
```

Comment:
Very Nit: in most places we prefix our callbacks with `handle`, like `handleAddressChange`, so that in the component we have `onAddressChange={handleAddressChange}`. I found it pretty elegant but we're not really consistent as we can see in this file. Do you think we should enforce this or we don't care?

Reply:
Thanks for catching this, I don't think it's too important to enforce it, but I'll do a cleanup in this PR and be more careful in the future. OTOH however, I like it when the callback has the same name as the prop because then it doesn't need to be written twice for example in a context value. I'll mention your name next time I find something "discussable" ;)

Reply:
What I don't like with `handle...` is to call it programmatically. I like it when I see it as a callback to buttons like
```tsx
<Button title="Generate" onClick="handleGenerateClick" />
```

but I don't like it when we need to call it as:

```tsx
const foo = () => {
  // doing some stuff here, blah blah blah
  // ...
  handleGenerateClick()
  // ...
}

<Button title="Generate" onClick="handleGenerateClick" />
}
```

In the latter case I prefer to see:

```tsx
const foo = () => {
  // doing some stuff here, blah blah blah
  // ...
  onGenerateClick()
  // ...
}

<Button title="Generate" onClick="onGenerateClick" />
}
```

WDYT?

Same goes for `onClose`

Reply:
(Rename was done in 212f3b421)

Reply:
Yep I agree with you fully.